core/src/main/java/org/ehcache/core/config/SizedResourcePoolImpl.java

```diff
@@ -74,10 +68,25 @@ public ResourceUnit getUnit() {
 
   /**
    * {@inheritDoc}
+   *
+   * @throws IllegalArgumentException {@inheritDoc}
    */
   @Override
-  public boolean isPersistent() {
-    return persistent;
+  public void validateUpdate(final ResourcePool newPool) {
+    super.validateUpdate(newPool);
+
+    SizedResourcePool sizedPool = (SizedResourcePool)newPool;
+    // Ensure unit type has not changed
+    if (!this.getUnit().getClass().equals(sizedPool.getUnit().getClass())) {
```

Comment:
The old code in `org.ehcache.core.config.ResourcePoolsImpl#validateAndMerge` prevented any unit change -- even among `MemoryUnit` constants.  After a brief discussion with @chrisdennis, I switched to this method to prevent switching between `MemoryUnit` and `EntryUnit`.